test/test_functional.py

```diff
@@ -0,0 +1,182 @@
+
+    def test_istft_is_inverse_of_stft4(self):
+        # hamming_window, not centered, not normalized, onesided
+        # window same size as n_fft
+        kwargs4 = {
+            'n_fft': 5,
+            'hop_length': 2,
+            'win_length': 5,
+            'window': torch.hamming_window(5),
+            'center': False,
+            'pad_mode': 'constant',
+            'normalized': False,
+            'onesided': True,
+        }
+
+        self._test_istft_is_inverse_of_stft(kwargs4)
+
+    def test_istft_is_inverse_of_stft5(self):
+        # hamming_window, not centered, not normalized, not onesided
+        # window same size as n_fft
+        kwargs5 = {
+            'n_fft': 3,
+            'hop_length': 2,
+            'win_length': 3,
+            'window': torch.hamming_window(3),
+            'center': False,
+            'pad_mode': 'reflect',
+            'normalized': False,
+            'onesided': False,
+        }
+
+        self._test_istft_is_inverse_of_stft(kwargs5)
+
+    def test_istft_of_ones(self):
+        # stft = torch.stft(torch.ones(4), 4)
+        stft = torch.tensor([
+            [[4., 0.], [4., 0.], [4., 0.], [4., 0.], [4., 0.]],
+            [[0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.]],
+            [[0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.]]
+        ])
+
+        estimate = torchaudio.functional.istft(stft, n_fft=4, length=4)
+        self._compare_estimate(torch.ones(4), estimate)
+
+    def test_istft_of_zeros(self):
+        # stft = torch.stft(torch.zeros(4), 4)
+        stft = torch.zeros((3, 5, 2))
+
+        estimate = torchaudio.functional.istft(stft, n_fft=4, length=4)
+        self._compare_estimate(torch.zeros(4), estimate)
+
+    def test_istft_requires_overlap_windows(self):
+        # the window is size 1 but it hops 20 so there is a gap which throw an error
+        stft = torch.rand((3, 5, 2))
+        self.assertRaises(AssertionError, torchaudio.functional.istft, stft, n_fft=4,
+                          hop_length=20, win_length=1, window=torch.ones(1))
+
+    def test_istft_requires_nola(self):
+        stft = torch.zeros((3, 5, 2))
+        kwargs_ok = {
+            'n_fft': 4,
+            'win_length': 4,
+            'window': torch.ones(4),
+        }
+
+        kwargs_not_ok = {
+            'n_fft': 4,
+            'win_length': 4,
+            'window': torch.zeros(4),
+        }
+
+        # A window of ones meets NOLA but a window of zeros does not. This should
+        # throw an error.
+        torchaudio.functional.istft(stft, **kwargs_ok)
+        self.assertRaises(AssertionError, torchaudio.functional.istft, stft, **kwargs_not_ok)
+
+    def _test_istft_of_sine(self, amplitude, L, n):
+        # stft of amplitude*sin(2*pi/L*n*x) with the hop length and window size equaling L
+        x = torch.arange(2 * L, dtype=torch.get_default_dtype())
+        sound = amplitude * torch.sin(2 * math.pi / L * x * n)
```

Comment:
The way you have set it up, L here is your number of point, so I would remove it from the solution: `sin(2 pi n x)`, and this way it will not be needed in the definition of `stft_largest_val`.

Reply:
I'm just going to keep it as it describes how long the window function is